File: Python/day2/day2.py
part1_results = {
    'A X': 4,
    'A Y': 8,
    'A Z': 3,
    'B X': 1,
    'B Y': 5,
    'B Z': 9,
    'C X': 7,
    'C Y': 2,
    'C Z': 6
}

part2_results = {
    'A X': 3,
    'A Y': 4,
    'A Z': 8,
    'B X': 1,
    'B Y': 5,
    'B Z': 9,
    'C X': 2,
    'C Y': 6,
    'C Z': 7
}

def part1(inputs):
    score = 0
    for input in inputs:
        score += part1_results[input]

    return score

def part2(inputs):
    score = 0
    for input in inputs:
        score += part2_results[input]

    return score



if __name__ == "__main__":
    train = [line.strip() for line in open("Python/day2/train.txt", "r")]
    test = [line.strip() for line in open("Python/day2/test.txt", "r")]

    assert part1(train) == 15
    assert part1(test) == 8933
    assert part2(train) == 12
    assert part2(test) == 11998
    
    print(f"Train Part 1: {part1(train)}")
    print(f"Test Part 1: {part1(test)}")
    print(f"Train Part 2: {part2(train)}")
    print(f"Test Part 1: {part2(test)}")

File: elixiraoc/lib/day2.ex
defmodule Day2 do
  def test do
    IO.puts("Train part1: #{part1("inputs/day2/train.txt")}")
    IO.puts("Test part1: #{part1("inputs/day2/test.txt")}")
    IO.puts("Train part2: #{part2("inputs/day2/train.txt")}")
    IO.puts("Test part2: #{part2("inputs/day2/test.txt")}")
  end

  def part1(path), do: input(path) |> Enum.map(&solve/1) |> Enum.sum()
  def part2(path), do: input(path) |> Enum.map(&solve2/1) |> Enum.sum()

  def input(path), do: path |> File.stream!() |> Enum.map(&parse_input/1)

  def parse_input(input), do: input |> String.trim()

  def solve("A Y"), do: 8
  def solve("A X"), do: 4
  def solve("A Z"), do: 3
  def solve("B Y"), do: 5
  def solve("B X"), do: 1
  def solve("B Z"), do: 9
  def solve("C Y"), do: 2
  def solve("C X"), do: 7
  def solve("C Z"), do: 6
  def solve2("A Y"), do: 4
  def solve2("A X"), do: 3
  def solve2("A Z"), do: 8
  def solve2("B Y"), do: 5
  def solve2("B X"), do: 1
  def solve2("B Z"), do: 9
  def solve2("C Y"), do: 6
  def solve2("C X"), do: 2
  def solve2("C Z"), do: 7
end
